src/test/java/core/basesyntax/service/WriterTest.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.service;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class WriterTest {
+
+    private static final String WRITER_RESULT_PATH = "src/test/java/core/basesyntax/resources/writerResult.txt";
+    private static final String WRONG_PATH = "src/test/java/core/basesyntax/resources";
+
+    @Test
+    void writeToFile_emptyReportProvided() {
+        Writer.writeToFile(WRITER_RESULT_PATH, "");
+
+        int actualLinesQuantity = Reader.readLines(WRITER_RESULT_PATH).size();
+        int expectedLinesQuantity = 0;
+
+        assertEquals(expectedLinesQuantity, actualLinesQuantity);
+
+    }
+
+    @Test
+    void writeToFile_wrongPathProvided_failure() {
+        assertThrows(FileNotFoundException.class, ()->Writer.writeToFile(WRONG_PATH, ""));
+    }
+
+    @Test
+    void writeToFile_rightReportProvided() {
+        Writer.writeToFile(WRITER_RESULT_PATH, "fruit,quantity\nbanana,152\napple,90");
+
+        List<String> writerResult = Reader.readLines(WRITER_RESULT_PATH);
+
+        int actualLinesQuantity = writerResult.size();
+        int expectedLinesQuantity = 2;
+
+        String actualFirstLine = writerResult.get(0);
+        String expectedFirstLine = "banana,152";
+
+        String actualSecondLine = writerResult.get(1);
+        String expectedSecondLine = "apple,90";
+
+        assertEquals(expectedLinesQuantity, actualLinesQuantity);
+        assertEquals(expectedFirstLine, actualFirstLine);
+        assertEquals(expectedSecondLine, actualSecondLine);
+    }
```

Comment:
Let's imagine that provided report is incorrect and we will have ```actualLinesQuantity``` not 2, for example 1.
Then, if we try to get ```actualSecondLine``` ```writerResult.get(1)``` will throw ```IndexOutOfBoundsException``` before we execute ```assertEquals``` cheks.
That's why I would recommend to do the checks after each expected/actual pair, like this:
```suggestion
    @Test
    void writeToFile_rightReportProvided() {
        Writer.writeToFile(WRITER_RESULT_PATH, "fruit,quantity\nbanana,152\napple,90");

        List<String> writerResult = Reader.readLines(WRITER_RESULT_PATH);

        int actualLinesQuantity = writerResult.size();
        int expectedLinesQuantity = 2;
        assertEquals(expectedLinesQuantity, actualLinesQuantity);

        String actualFirstLine = writerResult.get(0);
        String expectedFirstLine = "banana,152";
        assertEquals(expectedFirstLine, actualFirstLine);

        String actualSecondLine = writerResult.get(1);
        String expectedSecondLine = "apple,90";
        assertEquals(expectedSecondLine, actualSecondLine);
    }
```

Reply:
Thanks for the wonderful advice! Done.